snowfall/training/ctc_graph.py

```diff
@@ -69,8 +67,9 @@ def __init__(self,
         self.phones = phones
         self.words = words
         self.oov = oov
-        ctc_topo_inv = build_ctc_topo(list(phones._id2sym.keys())).invert_()
-        self.ctc_topo_inv = k2.arc_sort(ctc_topo_inv)
+        phone_ids = get_phone_symbols(phones)
```

Comment:
A difference here is that ctc_topo no longer includes the disambig symbols.
Of course it's not right to treat the disambig symbols as if they were phones.
But doesn't L contain disambig symbols on its input side?  Perhaps what is happening is that we are eliminating paths with disambig symbols.   We might need to add self-loop with each of the disambig symbols, on all of the states of the ctc topo.

Reply:
Now the L graph contain disambig symbols on its input side. That might be a problem in the current ctc topology. I think it makes sense to add  self-loop with each disambig symbols.

Reply:
I think I was mistaken about this L having disambig symbols, we are loading L.fst in train.py, which lacks them.

Reply:
I will check it. If there are no disambig symbols in L.fst, there may be problems while decoding.